cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -188,6 +189,11 @@ public static LoggingContext getLoggingContextWithRunId(String namespaceId, Stri
     return getLoggingContext(namespaceId, applicationId, entityId, programType, runId, systemArgs);
   }
 
+  public static LoggingContext getLoggingContextWithProgramRunId(ProgramRunId program, Map<String, String> systemArgs) {
```

Comment:
This method seems unnecessary. You are creating a `ProgramRunId` and then call this method in `AbstractDistributedProgramRunner`, however, you can just call the other `getLoggingContext` method directly.